Feature/new trip

AsherSpurr opened the pull request:
# Pull Request

## Description
Updates test file to test functions 

## Related Issues
N/A

## Changes Made
**scripts.js file now includes:**
- renderTripSelect -- shows destination options to the DOM on the select element
- createTrip -- will create a trip object from the plan form to POST
- userId & allTrips global variables added
- estimateCost -- estimated price of the trip that was most recently added

**apiCalls.js file now includes:**
- POST fetch request for a new trip
- unfinished DELETE fetch request

**domUpdates.js file now includes:**
- hide & unhide of modal 
- hide and unhide of main page
- enables the plan button -- hides the main page and opens up the trip form
- add home button -- enables to hide the form and unhide home
- checkIfSelected -- handles the select element of the trip form
- submitButton -- event listener grabs appropriate data
- in submitButton event listener -- createTrip from scripts.js is called
- renderEstimatedCost -- shows a trips estimated cost

index.HTML file now includes
- login page 
- login form

## Checklist- 
[x] The code follows the project's coding standards.
[ ]Unit tests have been added or updated to cover the changes.
[x] Documentation has been updated to reflect the changes (if applicable).
[x] The code compiles without errors.
[x] The changes have been tested locally and pass all relevant tests.
[ ] All new and existing tests pass.
[x] The pull request has been reviewed by at least one other contributor.

AsherSpurr committed:
feature: add new trip form; update DOM with dest options

AsherSpurr committed:
feature: unfinished/form inputs from user

AsherSpurr committed:
feature: create new trip from form input

AsherSpurr committed:
feature: unhide plan trip form on event

AsherSpurr committed:
feature: add page overlay on modal unhide

AsherSpurr committed:
feature: add login to HTML and CSS

AsherSpurr committed:
feature: add estimated cost function

dkwon1223 commented:
Asher, great job with your functions! They all seem very modular and follow the SRP expectations. I also like how you're implementing a hidden class to display and hide the login form vs the main page as I'm also taking a similar approach. I love the clarity and descriptive breakdown of your PR template. The only feedback I have for you is the following: 
- Despite the description, as I'm going through all changes and changed files, I'm not seeing any test files. Are you testing your functions as you're going? Curious to see your approach.
- The instructors did mention that these acountabilibuddy PR's are supposed to be something fairly small/one area of focus for us to review for each other. I'm definitely following all the commits for this PR but there's a lot going on and I'm not sure what you want me to be focusing on. Maybe this would be something to include in your next PR for me! I'll be sure to do it for you as well. 
- This may come off nitpicky and I certainly don't think it's an issue but if you'd like your checklist to appear as a checkbox in your PR template, you need to add a dash - before the [ ] brackets and it will look more "checkboxy" LOL. 

I appreciate all your effort and I'm glad to have you as an accountabilibuddy as this has definitely given me some things to think about for my own project and how I plan on giving you PR's to reveiw! Thank you!

AsherSpurr commented:
> Asher, great job with your functions! They all seem very modular and follow the SRP expectations. I also like how you're implementing a hidden class to display and hide the login form vs the main page as I'm also taking a similar approach. I love the clarity and descriptive breakdown of your PR template. The only feedback I have for you is the following:
> 
> * Despite the description, as I'm going through all changes and changed files, I'm not seeing any test files. Are you testing your functions as you're going? Curious to see your approach.
> * The instructors did mention that these acountabilibuddy PR's are supposed to be something fairly small/one area of focus for us to review for each other. I'm definitely following all the commits for this PR but there's a lot going on and I'm not sure what you want me to be focusing on. Maybe this would be something to include in your next PR for me! I'll be sure to do it for you as well.
> * This may come off nitpicky and I certainly don't think it's an issue but if you'd like your checklist to appear as a checkbox in your PR template, you need to add a dash - before the [ ] brackets and it will look more "checkboxy" LOL.
> 
> I appreciate all your effort and I'm glad to have you as an accountabilibuddy as this has definitely given me some things to think about for my own project and how I plan on giving you PR's to reveiw! Thank you!

Thank you for this comment! For the next PR I will give more direction and add reviewer instructions to let you know what to look out for.
- I am testing but haven't written tests for this branch yet. I have been putting the tests on separate branches. 
- For the next PR I will absolutely take the size into account. I completely spaced and did not realize just how much I had added. So thank you big time for taking the time to go through it!
- I love nitpicky because I didn't even realize my checkboxes weren't showing up, thank you for pointing that out!
